cryptofeed/exchange/bitcoincom.py

```diff
@@ -6,134 +6,40 @@
-            BID: sd({
-                Decimal(bid['price']): Decimal(bid['size']) for bid in msg['bid']
-            }),
-            ASK: sd({
-                Decimal(ask['price']): Decimal(ask['size']) for ask in msg['ask']
-            })
+        urls = {
+            'market': 'wss://api.exchange.bitcoin.com/api/2/ws/public',
+            'trading': 'wss://api.exchange.bitcoin.com/api/2/ws/trading',
+            'account': 'wss://api.exchange.bitcoin.com/api/2/ws/account',
         }
-        await self.book_callback(self.l2_book[pair], L2_BOOK, pair, True, None, timestamp_normalize(self.id, msg['timestamp']), timestamp)
-
-    async def _book_update(self, msg: dict, timestamp: float):
-        delta = {BID: [], ASK: []}
-        pair = self.exchange_symbol_to_std_symbol(msg['symbol'])
-        for side in ('bid', 'ask'):
-            s = BID if side == 'bid' else ASK
-            for entry in msg[side]:
-                price = Decimal(entry['price'])
-                amount = Decimal(entry['size'])
-                if amount == 0:
-                    delta[s].append((price, 0))
-                    del self.l2_book[pair][s][price]
-                else:
-                    delta[s].append((price, amount))
-                    self.l2_book[pair][s][price] = amount
-        await self.book_callback(self.l2_book[pair], L2_BOOK, pair, False, delta, timestamp_normalize(self.id, msg['timestamp']), timestamp)
-
-    async def message_handler(self, msg: str, conn, timestamp: float):
-
-        msg = json.loads(msg, parse_float=Decimal)
-        if 'result' in msg and msg['result'] is True:
-            return
-        elif 'method' in msg:
-            data = msg['params']
-
-            if 'sequence' in data:
-                if data['symbol'] not in self.seq_no:
-                    self.seq_no[data['symbol']] = data['sequence']
-                elif self.seq_no[data['symbol']] + 1 != data['sequence']:
-                    LOG.warning("%s: missing sequence number. Received %d, expected %d", self.id, data['sequence'],
-                                self.seq_no[data['symbol']] + 1)
-                    raise MissingSequenceNumber
-                self.seq_no[data['symbol']] = data['sequence']
-
-            if msg['method'] == 'snapshotOrderbook':
-                await self._book_snapshot(data, timestamp)
-            elif msg['method'] == 'updateOrderbook':
-                await self._book_update(data, timestamp)
-            elif msg['method'] == 'snapshotTrades':
-                return
-            elif msg['method'] == 'updateTrades':
-                await self._trade(data, timestamp)
-            elif msg['method'] == 'ticker':
-                await self._ticker(data, timestamp)
-            else:
-                LOG.warning("%s: Invalid message type %s", self.id, msg)
-        else:
-            LOG.warning("%s: Invalid message type %s", self.id, msg)
+        super().__init__(candle_interval='1m')
```

Comment:
I'm not sure you need both supers here, I'm pretty sure you just need

`super().__init__(urls, candle_interval=candle_interval, **kwargs)`

Also, candle_interval should be a kwarg in the constructor here

Reply:
I'm going to attempt to reply like I know what I'm talking about... ;)

The second super has Bequant as its subclass argument. This means it is calling the __init__ method from the class _above_ Bequant, i.e. Feed. And this is necessary because the urls need to be passed to Feed, not to Bequant.

I think I probably shouldn't have also passed urls to Bequant's __init__. I can remove these. But they are having no effect anyway.

Reply:
the bequant class already calls super on the parent class (Feed). Look at okex or any other the others that subclass. you don't need to call init twice.

Reply:
OK, thanks for the pointer. I'll take a look tomorrow

Reply:
Yep, of course you were right. I've reworked it.

Re. candle_interval, I removed it altogether as this all gets inherited from parent and it was creating multiple assignments.